packages/rpc-graphql/src/resolvers/__tests__/account-resolver-test.ts

```diff
@@ -0,0 +1,83 @@
+import type { Rpc } from '@solana/rpc-types';
+import fetchMock from 'jest-fetch-mock-fork';
+
+import { createRpcGraphQL, RpcGraphQL } from '../../index';
+
+describe('account resolver', () => {
+    let rpc: Rpc<GetAccountInfoApi & GetBlockApi & GetMultipleAccountsApi & GetProgramAccountsApi & GetTransactionApi>;
+    let rpcGraphQL: RpcGraphQL;
+    beforeEach(() => {
+        fetchMock.resetMocks();
+        fetchMock.dontMock();
+        rpc = createJsonRpc<
+            GetAccountInfoApi & GetBlockApi & GetMultipleAccountsApi & GetProgramAccountsApi & GetTransactionApi
+        >({
+            api: createSolanaRpcApi(),
+            transport: createHttpTransport({ url: 'http://127.0.0.1:8899' }),
+        });
+        rpcGraphQL = createRpcGraphQL(rpc);
+    });
+
+    describe('address-only requests', () => {
+        describe('in the first level', () => {
+            it('will not call the RPC for only an address', async () => {
+                expect.assertions(1);
+                const source = /* GraphQL */ `
+                    query testQuery {
+                        account(address: "2JPQuT3dHtPjrdcbUQyrrT4XYRYaWpWfmAJ54SUapg6n") {
+                            address
+                        }
+                    }
+                `;
+                await rpcGraphQL.query(source);
+                expect(fetchMock).not.toHaveBeenCalled();
+            });
+        });
+        describe('in the second level', () => {
+            it('will not call the RPC for only an address', async () => {
+                expect.assertions(1);
+                const source = /* GraphQL */ `
+                    query testQuery {
+                        account(address: "CSg2vQGbnwWdSyJpwK4i3qGfB6FebaV3xQTx4U1MbixN") {
+                            address
+                            ownerProgram {
+                                address
+                            }
+                        }
+                    }
+                `;
+                await rpcGraphQL.query(source);
+                expect(fetchMock).toHaveBeenCalledTimes(1);
+            });
+        });
+        describe('in the third level', () => {
+            it('will not call the RPC for only an address', async () => {
+                expect.assertions(1);
+                const source = /* GraphQL */ `
+                    query testQuery {
+                        account(address: "AyGCwnwxQMCqaU4ixReHt8h5W4dwmxU7eM3BEQBdWVca") {
+                            address
+                            ownerProgram {
+                                address
+                                ownerProgram {
+                                    address
+                                }
+                            }
+                        }
+                    }
+                `;
+                await rpcGraphQL.query(source);
+                expect(fetchMock).toHaveBeenCalledTimes(2);
+            });
```

Comment:
Currently I'm struggling to figure out how to make it work with this one. It seems that, because the 2nd layer's fetch depends on the resolved promise from the 1st, and the 3rd layer's fetch depends on the resolved promise from the 2nd, this gets tricky to do with `fetchMock.mockOnce(..)` and `jest.runAllTimersAsync()`.

If you have insight here, let me know.